src/components/settings/SignerAccountMFA/PasswordForm.tsx

```diff
@@ -27,80 +62,187 @@ export const PasswordForm = ({ mpcCoreKit }: { mpcCoreKit: Web3AuthMPCCoreKit })
+          <Accordion defaultExpanded={isPasswordSet}>
+            <AccordionSummary>
+              <Box display="flex" alignItems="center" gap={1}>
+                <SvgIcon component={CheckIcon} sx={{ color: isPasswordSet ? 'success.main' : 'border.light' }} />
+                <Typography fontWeight="bold">Password</Typography>
+              </Box>
+            </AccordionSummary>
+            <AccordionDetails sx={{ p: 0 }}>
+              <Grid container>
+                <Grid item container xs={12} md={7} gap={3} p={3}>
+                  {isPasswordSet && (
+                    <>
+                      <Typography>You already have a recovery password setup.</Typography>
+                      <FormControl fullWidth>
+                        <PasswordInput
+                          name={PasswordFieldNames.oldPassword}
+                          placeholder="Old password"
+                          label="Old password"
+                          helperText={formState.errors[PasswordFieldNames.oldPassword]?.message}
+                          required
+                        />
+                      </FormControl>
+                      <Divider sx={{ width: 1 }} />
+                    </>
+                  )}
+
+                  <FormControl fullWidth>
+                    <PasswordInput
+                      name={PasswordFieldNames.newPassword}
+                      placeholder="New password"
+                      label="New password"
+                      helperText={formState.errors[PasswordFieldNames.newPassword]?.message}
+                      required
+                      inputProps={{
+                        onChange: (event: ChangeEvent<HTMLInputElement>) => {
+                          const confirmNewPW = getValues(PasswordFieldNames.confirmPassword)
+                          const value = event.target.value
+                          setPasswordsMatch(value !== '' && value === confirmNewPW)
+
+                          if (strongPassword.test(value)) {
+                            setPasswordStrength(PasswordStrength.strong)
+                          } else if (mediumPassword.test(value)) {
+                            setPasswordStrength(PasswordStrength.medium)
+                          } else {
+                            setPasswordStrength(PasswordStrength.weak)
+                          }
+                        },
+                      }}
+                    />
+                    <Typography
+                      variant="body2"
+                      display="flex"
+                      alignItems="center"
+                      gap={1}
+                      mt={1}
+                      className={
+                        passwordStrength === PasswordStrength.strong
+                          ? css.strongPassword
+                          : passwordStrength === PasswordStrength.medium
+                          ? css.mediumPassword
+                          : css.weakPassword
```

Comment:
We have two big condition blocks like this: 
For the css class and for the label, should we add a `Record<PasswordStrength, { label: string, cssClass: string}>` mapping?